libraries/AP_OSD/AP_OSD.cpp

```diff
@@ -599,6 +606,32 @@ void AP_OSD::update_current_screen()
             last_switch_ms = 0;
         }
         break;
+    case STICKS_INPUT:
+        if (!AP_Notify::flags.armed && !AP::runcam()->in_menu()) {
```

Comment:
You don't need to check for armed here. You can't be in menu and armed

Reply:
... also the wrong way to check if you're armed; you ask the arming library directly if you're armed.

Reply:
> You don't need to check for armed here. You can't be in menu and armed

Disagree.  We should not pre-suppose the inner-workings of AP_RunCam here.

We should both check we are not armed *and* that we aren't going to screw over AP_RunCam somehow.
```suggestion
        if (!AP::arming().is_armed() && !AP::runcam()->in_menu()) {
```

Reply:
... actually, you will need to add `#if`s around the call to the runcam singleton here, or we'll end up with compilations failures when runcam isn't compiled in